### PR TITLE
[webgpu] Use uniform for program batchNorm, clip, im2col

### PR DESCRIPTION
**Pull request by axinging**
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4958)

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @xhcao @haoyunfeix @gyagp, PTAL

File: tfjs-backend-webgpu/src/kernels/im2col_webgpu.ts [hunk 1]
```diff
@@ -44,34 +45,14 @@ export class Im2ColProgram implements WebGPUProgram {
         this.dispatchLayout, this.outputShape, this.workGroupSize,
         [this.workPerThread, 1, 1]);
     this.inputShape = inputShape;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/im2col_webgpu.ts:47]: nit: It seems that `this.inputShape` and `this.rank` are not used. Remove them?

File: tfjs-backend-webgpu/src/kernels/clip_webgpu.ts [hunk 2]
```diff
@@ -25,25 +25,21 @@ export class ClipProgram implements WebGPUProgram {
...
   size: number;
 
-  constructor(outputShape: number[], minVal: number, maxVal: number) {
+  constructor(outputShape: number[]) {
     this.outputShape = outputShape;
     this.dispatchLayout = flatDispatchLayout(this.outputShape);
     this.dispatch = computeDispatch(
-        this.dispatchLayout, this.outputShape, this.workGroupSize,
-        [this.workPerThread, 1, 1]);
+        this.dispatchLayout, this.outputShape, this.workGroupSize, [1, 1, 1]);
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/clip_webgpu.ts:40]: nit: ` this.dispatch = computeDispatch(this.dispatchLayout, this.outputShape, this.workGroupSize);`

File: tfjs-backend-webgpu/src/kernels/im2col_webgpu.ts [hunk 3]
```diff
@@ -86,16 +67,15 @@ export class Im2ColProgram implements WebGPUProgram {
...
+            int d0 = offsetY + dilation[1] * (pos / itemsPerBlockRow);
             float value = 0.0;
             if(d0 < aShape[${rowDim}] && d0 >= 0) {
-              int offsetX = int(mod(float(blockIndex), ${outWidth}.) *
-                ${strideWidth}. - ${left}.);
-              int d1 = offsetX + ${dilationWidth} * (int(mod(float(pos),
-                ${itemsPerBlockRow}.) / ${inChannels}.));
-              int ch = int(mod(float(pos), ${inChannels}.));
+              int offsetX = int(mod(float(blockIndex), float(outWidth)) *
+              float(stride[0]) - float(pad[0]));
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/im2col_webgpu.ts:75]: Why do we need to convert it to `float`? Why not directly use `int offsetX = mod(blockIndex, outWidth) * stride[0] - pad[0];`?

Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/im2col_webgpu.ts]: Remove float conversion is done

File: tfjs-backend-webgpu/src/kernels/clip_webgpu.ts (hunk 2 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/clip_webgpu.ts]: done

File: tfjs-backend-webgpu/src/kernels/im2col_webgpu.ts (hunk 1 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/im2col_webgpu.ts]: done

**Review by qjia7 — APPROVED** [review]
@lina128 @jinjingforever @kainino0x take a look, thanks.

**Review by jinjingforever — APPROVED** [review]

**Review by kainino0x — APPROVED** [review]
